src/assets/scripts/client/InputController.js

```diff
@@ -722,51 +563,98 @@ export default class InputController {
+                // this.$commandInput.css({ 'background-color': '#999' });
+                this.$commandInput.css({ color: 'white' });
+
+                return;
+
+            default:
+                return;
+        }
+    }
+
+    /**
+     * Process user command to be applied to an aircraft
+     *
+     * @for InputController
+     * @method processAircraftCommand
+     */
+    processAircraftCommand() {
+        //     this.input.history.unshift(this.input.callsign);
+        //     this.$commandInput.val('');
+        //     this.input.command = '';
+
+        let aircraftCommandParser;
         // this could use $commandInput.val() as an alternative
         const userCommand = this.input.command.trim().toLowerCase();
 
-        // Using try/catch here very much on purpose. the `CommandParser` will throw when it encounters any kind
+        // Using try/catch here very much on purpose. the `AircraftCommandParser` will throw when it encounters any kind
         // of error; invalid length, validation, parse, etc. Here we catch those errors, log them to the screen
         // and then throw them all at once
         try {
-            result = new CommandParser(userCommand);
+            aircraftCommandParser = new AircraftCommandParser(userCommand);
         } catch (error) {
             UiController.ui_log('Command not understood');
 
             throw error;
         }
 
-        return result;
+        if (aircraftCommandParser.command !== 'transmit') {
+            return this.processSystemCommand(aircraftCommandParser);
+        }
+
+        return this.processTransmitCommand(aircraftCommandParser);
     }
 
     /**
+     * Process user command to be applied to the user's scope
+     *
      * @for InputController
-     * @method input_run
+     * @method processScopeCommand
      */
-    input_run() {
-        const commandParser = this._parseUserCommand();
+    processScopeCommand() {
+        let scopeCommandModel;
+        const userCommand = this.input.command.trim().toLowerCase();
 
-        if (commandParser.command !== 'transmit') {
-            return this.processSystemCommand(commandParser);
+        try {
+            scopeCommandModel = new ScopeCommandModel(userCommand);
+        } catch (error) {
+            UiController.ui_log('ERROR: BAD SYNTAX');
+
+            throw error;
         }
 
-        return this.processTransmitCommand(commandParser);
+        const [successful, response] = this._scopeModel.runScopeCommand(scopeCommandModel);
+
+        UiController.ui_log(response, !successful);
     }
 
     /**
      * @for InputController
      * @method processSystemCommand
-     * @param commandParser {CommandParser}
+     * @param AircraftCommandParser {AircraftCommandParser}
      * @return {boolean}
      */
-    processSystemCommand(commandParser) {
-        switch (commandParser.command) {
+    processSystemCommand(AircraftCommandParser) {
```

Comment:
this should be lowercase since it is and instance.

Reply:
GAH AUTOCOMPLETE!

done